extension/src/popup/SorobanContext.tsx

```diff
@@ -0,0 +1,61 @@
+import React from "react";
+import { useSelector } from "react-redux";
+import * as SorobanClient from "soroban-client";
+
+import { SOROBAN_RPC_URLS } from "@shared/constants/stellar";
+
+import { settingsNetworkDetailsSelector } from "./ducks/settings";
+
+export const SorobanContext = React.createContext({} as SorobanTxBuilder);
+
+export class SorobanTxBuilder {
+  public source: SorobanClient.Account;
+  private fee: string;
+  private pubKey: string;
+  private networkPassphrase: string;
+  public server: SorobanClient.Server;
+
+  constructor(pubKey: string, fee: string, networkPassphrase: string) {
+    this.fee = fee;
+    this.pubKey = pubKey;
+    this.networkPassphrase = networkPassphrase;
+
+    this.server = new SorobanClient.Server(SOROBAN_RPC_URLS.futureNet, {
+      allowHttp: SOROBAN_RPC_URLS.futureNet.startsWith("http://"),
+    });
```

Comment:
we should also make sure this works for local networks. The cheesy solution this might be something like:

```
if (networkPassphrase === "Test SDF Future Network ; October 2022" && networkUrl === "https://horizon-futurenet.stellar.org/" ) {
  serverUrl = SOROBAN_RPC_URLS.futureNet
} else {
  serverUrl = networkUrl; // in this case, localhost:8000/soroban/rpc
}

Reply:
Ah yeah good point, I can refactor to explicitly pass the `allowHttp` flag when I turn this into a function.

Reply:
Added in 16200e44ab98a1c820f7e795091a04b53b76c325